functions/Set-DbaAgentOperator.ps1

```diff
@@ -274,14 +274,13 @@ function Set-DbaAgentOperator {
 
         if ($SqlInstance) {
             try {
-                $InputObject += Get-DbaAgentOperator -SqlInstance $SqlInstance -SqlCredential $SqlCredential -Operator $($op.Name) -EnableException
+                $InputObject += Get-DbaAgentOperator -SqlInstance $SqlInstance -SqlCredential $SqlCredential -Operator $Operator -EnableException
             } catch {
                 Stop-Function -Message "Failed" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
             }
         }
 
         foreach ($op in $InputObject) {
-            $server = $op | Get-ConnectionParent
```

Comment:
You need $server in line 294 for the refresh of the SMO - why have you deleted this line?

Reply:
@potatoqualitee - Please add this line back in. Then this PR is ready for merge.

Reply:
will do, thanks @andreasjordan !